### PR TITLE
Bug 2040620: Rename MAPO image

### PR DESCRIPTION
**Pull request by mandre**
The image needed to change name to adhere to conventions.

Depends on https://github.com/openshift/release/pull/25351

### CODE REVIEW

**Comment by pierreprinetti** [comment]
/assign @joelspeed
as the bot suggests

**Comment by JoelSpeed** [comment]
/hold

I don't think the naming in the image reference matches the build data, this will need to be double checked

https://github.com/openshift/ocp-build-data/blob/openshift-4.10/images/ose-machine-api-provider-openstack.yml

File: install/image-references [hunk 1]
```diff
@@ -14,6 +14,10 @@ spec:
     from:
       kind: DockerImage
       name: registry.svc.ci.openshift.org/openshift:openstack-machine-controllers
+  - name: openstack-machine-api-provider
```
Comment (JoelSpeed) [inline install/image-references:17]: This string needs to match https://github.com/openshift/ocp-build-data/blob/b75c21e23e55142ea58b037d0d4c4e7878b33ba2/images/ose-machine-api-provider-openstack.yml#L25 once you strip the `openshift/ose-` prefix

Comment (mandre) [reply install/image-references:17]: It should be good now thanks to https://github.com/openshift/ocp-build-data/pull/1275.

Comment (mandre) [reply install/image-references:17]: The image is not yet available in the `4.10-art-latest` image stream though, I'll let you know when we can merge this.

Comment (mandre) [reply install/image-references:17]: It's good to go, the image is there now:
```
$ oc describe imagestream 4.10-art-latest | grep openstack-machine-api-provider
openstack-machine-api-provider
```

File: install/image-references [hunk 2]
```diff
@@ -14,6 +14,10 @@ spec:
...
+    from:
+      kind: DockerImage
+      name: quay.io/openshift/origin-openstack-machine-api-provider
```
Comment (mandre) [inline install/image-references:20]: I'm still not clear how this ref gets replaced by the one from the release payload, if that's what is really happening.

**Comment by JoelSpeed** [comment]
/override ci/prow/e2e-aws-upgrade

The test failures are unrelated to this PR. This PR does not affect AWS (apart from image build) and as such, given the cluster bootstrapped for the various tasks this clearly hasn't broken anything. Especially as the new image reference is only used with tech preview clusters, this test doesn't make sense here

**Pull request merged**